core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -1608,15 +1634,20 @@ oppia.factory('explorationGadgetsService', [
 ]);
 
 // A service that returns the frontend representation of a newly-added state.
-oppia.factory('newStateTemplateService', [function() {
+oppia.factory('newStateTemplateService',
+  ['StateObjectFactory', function(StateObjectFactory) {
   return {
     // Returns a template for the new state with the given state name, changing
     // the default rule destination to the new state name in the process.
     // NB: clients should ensure that the desired state name is valid.
     getNewStateTemplate: function(newStateName) {
-      var newStateTemplate = angular.copy(GLOBALS.NEW_STATE_TEMPLATE);
-      newStateTemplate.interaction.default_outcome.dest = newStateName;
-      return newStateTemplate;
+      var newState = StateObjectFactory.create(newStateName, {
+        content: GLOBALS.NEW_STATE_TEMPLATE.content,
+        interaction: GLOBALS.NEW_STATE_TEMPLATE.interaction,
+        param_changes: GLOBALS.NEW_STATE_TEMPLATE.param_changes
+      });
+      newState.interaction.default_outcome.dest = newStateName;
```

Comment:
Make sure this doesn't change GLOBALS.NEW_STATE_TEMPLATE.interaction. Is there an angular.copy() somewhere in the pipeline?

Might want to angular.copy() on line 1646 anyway just to be safe, or construct the correct value for the 'interaction' first and only then pass it to the constructor. In fact, the latter is probably better -- mutations to these objects should really be done via setter methods.